Mathlib/MeasureTheory/Function/ConditionalExpectation/Basic.lean

```diff
@@ -387,34 +395,46 @@ lemma Memℒp.condExpL2_ae_eq_condExp (hm : m ≤ m₀) (hf : Memℒp f 2 μ) [I
     condExpL2 E 𝕜 hm hf.toLp =ᵐ[μ] μ[f | m] :=
   hf.condExpL2_ae_eq_condExp' hm (memℒp_one_iff_integrable.1 <| hf.mono_exponent one_le_two)
 
+end RCLike
+
+section Real
+variable [InnerProductSpace ℝ E]
+
 -- TODO: Generalize via the conditional Jensen inequality
-include 𝕜 in
 lemma eLpNorm_condExp_le : eLpNorm (μ[f | m]) 2 μ ≤ eLpNorm f 2 μ := by
   by_cases hm : m ≤ m₀; swap
   · simp [condExp_of_not_le hm]
   by_cases hfμ : SigmaFinite (μ.trim hm); swap
   · rw [condExp_of_not_sigmaFinite hm hfμ]
     simp
   by_cases hfi : Integrable f μ; swap
-  · rw [condExp_undef hfi]
+  · rw [condExp_of_not_integrable hfi]
     simp
   obtain hf | hf := eq_or_ne (eLpNorm f 2 μ) ∞
   · simp [hf]
   replace hf : Memℒp f 2 μ := ⟨hfi.1, Ne.lt_top' fun a ↦ hf (id (Eq.symm a))⟩
-  rw [← eLpNorm_congr_ae (hf.condExpL2_ae_eq_condExp' (𝕜 := 𝕜) hm hfi)]
+  rw [← eLpNorm_congr_ae (hf.condExpL2_ae_eq_condExp' (𝕜 := ℝ) hm hfi)]
   refine le_trans (eLpNorm_condExpL2_le hm _) ?_
   rw [eLpNorm_congr_ae hf.coeFn_toLp]
 
-include 𝕜 in
 protected lemma Memℒp.condExp (hf : Memℒp f 2 μ) : Memℒp (μ[f | m]) 2 μ := by
   by_cases hm : m ≤ m₀
   · exact ⟨(stronglyMeasurable_condExp.mono hm).aestronglyMeasurable,
-      eLpNorm_condExp_le (𝕜 := 𝕜).trans_lt hf.eLpNorm_lt_top⟩
+      eLpNorm_condExp_le.trans_lt hf.eLpNorm_lt_top⟩
   · simp [condExp_of_not_le hm]
 
-end MemL2
+end Real
 end NormedAddCommGroup
 
+section NormedRing
+variable {R : Type*} [NormedRing R] [NormedSpace ℝ R] [CompleteSpace R]
+
+lemma condExp_ofNat (n : ℕ) [n.AtLeastTwo] (f : α → R) :
+    μ[OfNat.ofNat n * f|m] =ᵐ[μ] OfNat.ofNat n * μ[f|m] := by
```

Comment:
```suggestion
    μ[ofNat(n) * f|m] =ᵐ[μ] ofNat(n) * μ[f|m] := by
```

Reply:
But this is not a simp lemma?

Reply:
Don't you want it to be a simp lemma?

Reply:
I don't really mind. I guess it could be one, yes